### ISSUE TITLE
Linters, anyone?

### ISSUE THREAD

**Issue by impinball** [lhorie/mithril.js#706]
I think it would be fantastic if Mithril was actually covered by a linter. I have no religious objection to any linter, as long as it parses JavaScript, but things like #662 are easily preventable with a linter.

Other things I've noticed:

- Lines spanning 150+ columns [(This line literally can't fit on my screen, for example)](https://github.com/lhorie/mithril.js/blob/next/mithril.js#L266)
- Very inconsistent semicolon usage (~50% of the statements have them)
- A few long and/or complex boolean expressions that probably could stand to be wrapped (ties in with the first point)
- A few cases where the [code is doing the work of the minifier](https://github.com/lhorie/mithril.js/blob/next/mithril.js#L249). UglifyJS is smart enough to know to minify many of these kinds of things (among others):

```js
// In
if (foo) {
  var bar = doA();
} else if (baz) {
  var bar = doB(baz);
} else {
  var bar = doC();
}

// Out
var bar=foo?doA():baz?doB(baz):doC();
```

**Comment by kaukas** [comment]
+1.

Actually +10! Automatic [JSHint](https://github.com/jshint/jshint/) and [JSCS](https://github.com/jscs-dev/node-jscs) checking helps clean up code and simplifies contributions. And please add [CodeClimate](https://codeclimate.com/oss) to that.

I do not think suggestions by these tools should be followed religiously. But having *some* agreed conventions for a widely used library is way better than having none.

Is help needed to set these up?

**Comment by impinball** [comment]
And another complete WTF: [a ternary using logical operators instead of a ternary operator](https://github.com/lhorie/mithril.js/blob/next/mithril.js#L919). I'm at a loss for words on this one...

**Comment by impinball** [comment]
@kaukas It would result in a long series of commits + quite a few PRs running into merge conflicts for the short term, and it's not a small operation. I'll ping @lhorie here.

**Comment by impinball** [comment]
And [these](https://github.com/lhorie/mithril.js/blob/next/tests/mithril-tests.js#L1838-L1846) [two](https://github.com/lhorie/mithril.js/blob/next/tests/mithril-tests.js#L1899-L1908) tests fail if a triple-equals is used.

**Comment by elektronik2k5** [comment]
+1

**Comment by impinball** [comment]
@lhorie said in the gitter room that he was planning on hooking up his rewrite (which *will* feature multiple breaking API changes) with JSHint.

**Comment by maranomynet** [comment]
+1 for linting!

**Comment by impinball** [comment]
Fixed with #822, #828.
